cms/djangoapps/contentstore/views/course.py

```diff
@@ -1136,22 +1136,23 @@ def grading_handler(request, course_key_string, grader_index=None):
     course_key = CourseKey.from_string(course_key_string)
     with modulestore().bulk_operations(course_key):
         course_module = get_course_and_check_access(course_key, request.user)
+        course_details = CourseGradingModel.fetch(course_key)
+        course_structure = _course_outline_json(request, course_module)
```

Comment:
Hi @pomegranited, We've been reviewing and trying to apply your suggested changes. We have some doubts with this [issue](https://github.com/proversity-org/edx-platform/compare/proversity/ginkgo.1-campus/CAM2-3...open-craft:jill/proversity/ginkgo.1-campus/CAM2-3?expand=1#diff-968532c6bba75e1f61d1e1ddb51d7842R1143),
Can you help us to undestand the purpose of passing graders as an argument to _course_outline_json() and how is that we are avoiding calling CourseGradingModel.fetch twice.

Thanks

Reply:
Hi @Squirrel18,

`CourseGradingModel.fetch` is called directly [here in `grading_handler`](https://github.com/edx-olive-oc/edx-platform/pull/9/files#diff-968532c6bba75e1f61d1e1ddb51d7842R1139).  Then, just below that, we call [`_course_outline_json`](https://github.com/edx-olive-oc/edx-platform/pull/9/files#diff-968532c6bba75e1f61d1e1ddb51d7842R1140), which calls [`create_xblock_info`](https://github.com/edx/edx-platform/blob/a5ab1250968725a6d25fd4ca2f44df9a39179d0f/cms/djangoapps/contentstore/views/course.py#L338-L345), which takes an optional `graders` parameter, which we're currently not using.

If that `graders` parameter is not passed in, then [`create_xblock_info` calls `CourseGradingModel.fetch`](https://github.com/edx/edx-platform/blob/a5ab1250968725a6d25fd4ca2f44df9a39179d0f/cms/djangoapps/contentstore/views/item.py#L1098-L1102) to retrieve them.

So I've suggested [passing the graders to `_course_outline_json`](https://github.com/proversity-org/edx-platform/compare/proversity/ginkgo.1-campus/CAM2-3...open-craft:jill/proversity/ginkgo.1-campus/CAM2-3?expand=1#diff-968532c6bba75e1f61d1e1ddb51d7842R1143), and then [through to `create_xblock_info`](https://github.com/proversity-org/edx-platform/compare/proversity/ginkgo.1-campus/CAM2-3...open-craft:jill/proversity/ginkgo.1-campus/CAM2-3?expand=1#diff-968532c6bba75e1f61d1e1ddb51d7842R341), since we don't want to fetch data twice.